apps/api/src/modules/notifications/dtos/providers/vcTwilio-data.dto.ts

```diff
@@ -0,0 +1,116 @@
+import { IsNotEmpty, IsOptional, ValidateIf } from 'class-validator';
+
+export class VcTwilioDataDto {
+  @IsNotEmpty()
+  from: string;
+
+  @IsNotEmpty()
+  to: string;
+
+  // Need atleast one of url, twiml for successful request
+  @IsOptional()
+  url?: string;
+
+  @IsOptional()
+  twiml?: string;
+
+  @ValidateIf((obj: VcTwilioDataDto) => !obj.url && !obj.twiml)
+  @IsNotEmpty({
+    message: 'Request must include either a "url" or "twiml" parameter for Twilio voice calls',
+  })
+  validateUrlOrTwiml: string; // This is a dummy property to apply the custom validation logic
```

Comment:
Did this approach not work?: https://github.com/OsmosysSoftware/osmo-x/blob/961d794fd64a78747a226ab3316a11687b9cf318/apps/api/src/modules/notifications/dtos/create-notification-attachment.dto.ts#L10
Why did we need to add a dummy property?

Reply:
Removed dummy property. Used the approach suggested in create-notification-attachment.dto